src/svg.c

```diff
@@ -69,24 +70,34 @@ int svg_to_ps(lua_State *L) {
         snprintf(color, 256, "%f %f %f rg ", r/256.0, g/256.0, b/256.0);
         output = safe_append(output, &output_l, &max_output, color);
 
-        strokeFillOper = 'f';
+        switch (shape->fillRule) {
+            case NSVG_FILLRULE_NONZERO:
+                strokeFillOper = "f"; break;
+            case NSVG_FILLRULE_EVENODD:
+            default:
+                strokeFillOper = "f*"; break;
+        }
+
         if (shape->stroke.type == NSVG_PAINT_COLOR) {
-          strokeFillOper = 'B';
+          strokeFillOper = "B";
         } else {
           if (output_l + 2 > max_output) {
+            max_output += 2;
```

Comment:
You're incrementing max_output and keeping the increment in the realloc() both here and with the +4 below. However, I don't (yet?) see how this could be causing the error.